src/NextGen/dataHandler/dataHandler.ts

```diff
@@ -0,0 +1,8 @@
+import { IMParticleComponent } from '../core/component';
+
+export interface IMParticleDataHandler extends IMParticleComponent {
+    // QUESTION: Shouldn't we be passing in a type for data instead of being "any"?
+    // QUESTION: Shouldn't immediateUpload be optional?
+    saveData(data: any, immediateUpload: boolean);
```

Comment:
Agreed, we should definitely have a type for `data`.

For immediateUpload, is it possible to set default parameter values in TS?

Reply:
Yes, we can set a default of `false` for `immediateUpload` in TS. I personally think we should do this. I'm cool with revising my PR if that's the direction we want to go.